File: OnlineStoreClient/src/app/models/parameters-metadata.ts
export class ParametersMetaData{
    parametersSearchValues: { [parameter: string]: string }
    parametersNames: { [parameter: string]: string }
    parameters: any
    orderByColumns: string[]
    dependentSearchValues: { parameter: string, dependentOnParameter: string, dependentOnValues: any[], searchValues: any[] }[]
}

File: Shared/RequestFeatures/ParametersMetaData.cs
using Shared.RequestFeatures.ParametersAttributes;
using Shared.RequestFeatures.ParametersModels;
using System.Dynamic;

namespace Shared.RequestFeatures
{
    public class ParametersMetaData
    {
        public ExpandoObject Parameters { get; set; } = new();
        public Dictionary<string, object[]> ParametersSearchValues = new();
        public Dictionary<string, string> ParametersNames = new();
        public List<DependentSearchParameter> DependentSearchValues = new();
        public List<string>? OrderByColumns { get; set; } = new();
    }
}
